libs/core/src/lib/carousel/carousel.component.ts

```diff
@@ -529,9 +529,10 @@ export class CarouselComponent implements OnInit, AfterContentInit, AfterViewIni
     /** @hidden */
     private _goToFirstItem(): void {
         if (
-            this.pageIndicatorsCountArray.length === 0 &&
-            this._carouselService.currentTransitionPx !== 0 &&
-            this.slides.length > 1
+            (this.pageIndicatorsCountArray.length === 0 &&
+                this._carouselService.currentTransitionPx !== 0 &&
+                this.slides.length > 1) ||
+            this.slides.length === this._visibleSlidesNumericCount
```

Comment:
If there are 4 slides behavior is still weird. I think in that case 2-4 slides should be shown.

https://user-images.githubusercontent.com/20265336/169266882-560e2ba3-6c3b-4654-b140-4cb62330cbd2.mov